### PR TITLE
Fix issue with Mutiple choice macros

### PR DESCRIPTION
**Pull request by ztolley**
Fix issue with Mutiple choice macros and form group macros not supporting groupClass due to mutating props

### CODE REVIEW

File: src/templates/_macros/form.njk [hunk 1]
```diff
@@ -211,8 +212,7 @@
         {{ 'has-error' if props.error }}
         {{ 'js-ConditionalSubfield' if isConditional }}
         {{ modifier }}
-        {{ props.class if props.class }}
```
Comment (teneightfive) [inline src/templates/_macros/form.njk]: Not sure I fully understand this change. It seems you're just removing the `if props.class`?

Do you have an example of the error before and after? Might help explain it better.

Comment (teneightfive) [inline src/templates/_macros/form.njk]: It also looks like the docs are out of date as they don't mention `props.class`.

Comment (ztolley) [inline src/templates/_macros/form.njk]: Before, if you passed a value to props.class to the macro it ignored it. Something about the line

`{% set children = caller() if caller else renderAsMacro(props.children) %}`

stripped the props object of the class property.

The change here removes the use of props.class and instead just inserts class, which is taken from props while it is still there.

Comment (teneightfive) [inline src/templates/_macros/form.njk]: Hmm, that doesn't feel like it's fixing the real problem.

Maybe look at renderAsMacro and see if its mutating something which is removing the `class` property from `props`?

Comment (ztolley) [inline src/templates/_macros/form.njk]: its not render as macro then but one of the macros it calls in turn .

We need more tests as the macros become more complex.

File: src/templates/_macros/form.njk [hunk 2]
```diff
@@ -201,6 +201,8 @@
   {% set isLabelHidden = props.isLabelHidden | default(false) %}
   {% set isConditional = props.condition %}
   {% set modifier = props.modifier | concat('') | reverse | join(' c-form-group--') if props.modifier %}
+
```
Comment (teneightfive) [inline src/templates/_macros/form.njk]: Maybe see if you can remove these noise changes?

File: src/templates/_macros/form.njk [hunk 3]
```diff
@@ -211,8 +213,7 @@
         {{ 'has-error' if props.error and not isLabelHidden }}
         {{ 'js-ConditionalSubfield' if isConditional }}
         {{ modifier }}
-        {{ props.class if props.class }}
-      "
+        {{ props.class if props.class }}"
```
Comment (teneightfive) [inline src/templates/_macros/form.njk]: Plus the noise here? Revert back to original version with `"` on next line.

File: test/unit/macros/form.test.js [hunk 4]
```diff
@@ -384,4 +384,21 @@ describe('Nunjucks form macros', () => {
       })
     })
   })
+
+  describe('Formgroup component', () => {
+    it('should allow a class to be specified for use in the form-grouo wrapper', () => {
```
Comment (teneightfive) [inline test/unit/macros/form.test.js]: `grouo` -> `group`

**Pull request merged**